dissect/target/plugins/apps/webserver/apache.py

```diff
@@ -187,65 +217,161 @@ class ApachePlugin(WebserverPlugin):
         "/etc/httpd/conf/httpd.conf",
         "/etc/httpd.conf",
     ]
+    DEFAULT_ENVVAR_PATHS = ["/etc/apache2/envvars", "/etc/sysconfig/httpd", "/etc/rc.conf"]
 
     def __init__(self, target: Target):
         super().__init__(target)
-        self.access_log_paths, self.error_log_paths = self.get_log_paths()
+        self.server_root = None
+        self.access_paths = set()
+        self.error_paths = set()
+        self.find_logs()
 
     def check_compatible(self) -> None:
-        if not len(self.access_log_paths) and not len(self.error_log_paths):
-            raise UnsupportedPluginError("No Apache directories found")
+        if not self.access_paths and not self.error_paths:
+            raise UnsupportedPluginError("No Apache log files found")
 
-    def get_log_paths(self) -> tuple[list[Path], list[Path]]:
-        """
-        Discover any present Apache log paths on the target system.
+        if self.target.os == OperatingSystem.CITRIX:
+            raise UnsupportedPluginError("Use the 'apps.webserver.citrix' apache plugin instead")
+
+    def find_logs(self) -> None:
+        """Discover any present Apache log paths on the target system.
+        Populates ``self.access_paths`` and ``self.error_paths``.
 
         References:
+            - https://httpd.apache.org/docs/2.4/logs.html
+            - https://httpd.apache.org/docs/2.4/mod/mod_log_config.html
             - https://www.cyberciti.biz/faq/apache-logs/
             - https://unix.stackexchange.com/a/269090
         """
 
-        access_log_paths = set()
-        error_log_paths = set()
-
         # Check if any well known default Apache log locations exist
         for log_dir, log_name in itertools.product(self.DEFAULT_LOG_DIRS, self.ACCESS_LOG_NAMES):
-            access_log_paths.update(self.target.fs.path(log_dir).glob(f"{log_name}*"))
+            self.access_paths.update(self.target.fs.path(log_dir).glob(f"*{log_name}*"))
 
         for log_dir, log_name in itertools.product(self.DEFAULT_LOG_DIRS, self.ERROR_LOG_NAMES):
-            error_log_paths.update(self.target.fs.path(log_dir).glob(f"{log_name}*"))
+            self.error_paths.update(self.target.fs.path(log_dir).glob(f"*{log_name}*"))
+
+        seen = set()
 
         # Check default Apache configs for CustomLog or ErrorLog directives
         for config in self.DEFAULT_CONFIG_PATHS:
-            if (path := self.target.fs.path(config)).exists():
-                for line in path.open("rt"):
-                    line = line.strip()
+            if (path := self.target.fs.path(config)).exists() and path not in seen:
+                self._process_conf_file(path)
+                seen.add(path)
+
+        # Check all .conf files inside the server root
+        if self.server_root:
+            for path in self.server_root.rglob("*.conf"):
+                if path not in seen:
+                    self._process_conf_file(path)
+                    seen.add(path)
```

Comment:
```suggestion
            if (path := self.target.fs.path(config)).exists() and path not in seen:
                self._process_conf_file(path, seen)

        # Check all .conf files inside the server root
        if self.server_root:
            for path in self.server_root.rglob("*.conf"):
                if path not in seen:
                    self._process_conf_file(path, seen)
```
Would this work? I think theoretically you would process the same file twice with this if you don't pass the `seen` here, if one of these base files has a reference back to itself (or another default file path).

Reply:
What about this one?

Reply:
There's a flaw in your current logic that allows for files to be processed twice, this fixes that.

Say you have the following configuration:
```
/etc/apache2/apache2.conf
    # Normal config
/etc/httpd.conf
    # References /etc/apache2/apache2.conf
```

With your current logic, `/etc/apache2/apache2.conf` will be processed twice.

Take this code as an example:

```python
def _process_conf_file(path, seen=None, content=None):
    seen = set() if seen is None else seen
    if path in seen:
        return
    seen.add(path)

    print("processing", path)

    for key, value in content.items():
        _process_conf_file(key, seen, value)


MOCK_DATA = {
    "/path/a": {
        "/path/a/sub": {},
        "/path/a/sub/sub": {},
    },
    "/path/b": {
        "/path/b/sub": {},
        "/path/a": {},
    },
}

for path, value in MOCK_DATA.items():
    _process_conf_file(path, None, value)
```

`content` is just to disregard processing and emulate a recursive parse of a next file. Running this prints:
```
processing /path/a
processing /path/a/sub
processing /path/a/sub/sub
processing /path/b
processing /path/b/sub
processing /path/a  # <-- processed twice
```
Simply changing the loop to this fixes it:
```python
seen = set()
for path, value in MOCK_DATA.items():
    _process_conf_file(path, seen, value)
```
```
processing /path/a
processing /path/a/sub
processing /path/a/sub/sub
processing /path/b
processing /path/b/sub
```

You can simplify it even further, there's no reason to check `path not in seen` in this function since you can let `_process_conf_file` handle that check. There's one change needed in `_process_conf_file` to make it work properly: `seen = set() if seen is None else seen`.